Feature/update pom

jweiser opened the pull request:
Reorganize and streamline the POM file. All details are in the commit messages.

jweiser committed:
adds .gitignore file

jweiser committed:
updates hamcrest matcher from contains to hasItem

jweiser committed:
removes unused PowerMock references

jweiser committed:
updates and reorganizes POM file

- adds tags for description, license, developers, scm
- updates URL tag value
- reorganizes POM order to Maven recommended order (https://maven.apache.org/developers/conventions/code.html)
- updates release-common-lib to v1.2.0
- replaces PowerMock dependency with plain Mockito
- adds/updates groupId and version tag for plugins
- adds plugin to build section for releasing new versions

jweiser committed:
converts tabs to spaces for consistency in POM file

SolomonShorser-OICR commented on src/test/java/org/reactome/release/BioModelsUtilitiesTester.java:
```diff
@@ -6,21 +6,16 @@
-@PrepareForTest({BioModelsUtilities.class})
-@PowerMockIgnore({"org.apache.logging.log4j.*", "javax.management.*", "javax.script.*",
-        "javax.xml.*", "com.sun.org.apache.xerces.*", "org.xml.sax.*", "com.sun.xml.*", "org.w3c.dom.*"})
+@RunWith(MockitoJUnitRunner.class)
```
These changes have all been tested, right?

jweiser replied:
Yup, I ran the tests locally and they all still passed with the changes.

SolomonShorser-OICR commented on src/test/java/org/reactome/release/ModelsTSVParserTester.java:
```diff
@@ -81,7 +81,7 @@ public void testFileContainsCorrectMapping() {
 
         // Tests pathway contains correct BioModels id
-        assertThat(pathwayToBioModelsIds.get("R-HSA-199418"), contains("BIOMD0000000175"));
+        assertThat(pathwayToBioModelsIds.get("R-HSA-199418"), hasItem("BIOMD0000000175"));
```
I'm curious why this was done. What's wrong with `contains`?

jweiser replied:
`contains` requires all elements in the collection be specified in order to match where `hasItem` matches for any of the items.

R-HSA_199418 has more than one BioModels identifier mapped to it, so `hasItem` is needed to match a single one.  Alternatively, we could provide all identifiers and keep `contains`.  That might be a good way to go, too.

cookersjs commented on pom.xml:
```diff
@@ -1,30 +1,99 @@
-    <url>http://maven.apache.org</url>
+    <description>
+        Insertion of BioModels identifiers cross-references
+        (data generated by https://github.com/reactome/biomodels-mapper) for Pathway instances
```
Is there another tag that denotes dependencies/repositories used, rather then needing to cite in the description in parantheses?

jweiser replied:
I'm not sure I understand what you mean by another tag?

jweiser replied:
It looks like the `parent` and `dependency` tags are used for relationships for other repos, but I don't see any tags that discuss related or dependent repositories not directly used by the source of the current project.

cookersjs commented on pom.xml:
```diff
@@ -36,43 +105,39 @@
                 </configuration>
             </plugin>
+
+            <!-- allows 'mvn release' command to create a new release of the project -->
```
Do we have documentation somewhere that describes how to do releases this way?

SolomonShorser-OICR replied:
Standard documentation for the maven release plugin is here: https://maven.apache.org/maven-release/maven-release-plugin/

jweiser replied:
I can embed the URL in the comment just so it's easy to lookup how to use the command and for anyone new to the project and maven.

jweiser replied:
Done: https://github.com/reactome/release-biomodels/pull/1/commits/3410122180d2c40637f273a34ae9078efec1387f

jweiser committed:
updates POM file

- adds https for URL tag
- adds UTF-8 source encoding property
- binds maven package phase to assembly:single goal
- updates maven plugin comments

jweiser committed:
adds maven-surefire-plugin to run tests by default with 'mvn test'
